src/lib/swapper/swappers/ThorchainSwapper/types.ts

```diff
@@ -126,9 +126,35 @@ export type ThorChainId =
   | ThorUtxoSupportedChainId
 
 type ThorNodeStatusResponseSuccess = {
-  // Non-exhaustive, the 'done' status is all we care about here
-  observed_tx: {
-    status?: 'done' | 'incomplete'
+  // Non-exhaustive. https://thornode.ninerealms.com/thorchain/doc/ for the full response type.
+  tx?: {
+    id: string
+    memo: string
+    chain: string
+    from_address: string
+    to_address: string
+  }
+  stages: {
+    inbound_observed: {
+      started?: boolean
+      final_count: number
+      completed: boolean
+    }
+    inbound_confirmation_counted?: {
+      completed: boolean
+    }
+    inbound_finalised?: {
+      completed: boolean
+    }
+    swap_status?: {
+      pending: boolean
+    }
+    swap_finalised?: {
+      completed: boolean
+    }
+    outbound_signed?: {
+      completed: boolean
+    }
```

Comment:
The optionals are accurate here, see screenshots:

- `inbound_observed` is the only stage that will be consistently defined, though `started` may be undefined:

![image](https://github.com/shapeshift/web/assets/17035424/b7091c80-6136-4252-9d26-c8828005c7a2)
![image](https://github.com/shapeshift/web/assets/17035424/f53d1184-d92f-4e3d-8351-ddc074d49276)
- All other stages are effectively potentially undefined in the earlier stages
- `tx` is added for future reference, but is also undefined in the earlier stages
- `out_txs` isn't part of this yet, though we may want to add it if iterating on this to take over THOR status and going end-to-end on the completion detection of the outbound Tx